pkg/dns/target.go

```diff
@@ -153,6 +153,70 @@ func (t *ClusterGatewayTarget) setGeo(defaultGeo GeoCode) {
 	t.Geo = &geoCode
 }
 
+func (t *MultiClusterGatewayTarget) RemoveUnhealthyGatewayAddresses(probes []*v1alpha1.DNSHealthCheckProbe, listener gatewayv1beta1.Listener) {
+
+	//If we have no probes we can't determine health so return unmodified
+	if len(probes) == 0 {
+		return
+	}
+
+	//Build a map of gateway addresses and their health status
+	gwAddressHealth := map[string]bool{}
+	allunhealthy := true
+	for _, cgt := range t.ClusterGatewayTargets {
+		for _, gwa := range cgt.GatewayAddresses {
+			gwProbes := getProbeForGatewayAddress(probes, gwa, t.Gateway.Name, string(listener.Name))
```

Comment:
Why would we ever have more than one probe for a gateway/listeenr/address?

Reply:
we shouldn't, probably being too cautious here. i'll make an update. 👍

Reply:
Since we are using the name as the id, i doubt you can really. That said, maybe you could have multiple probes eventually. 

If you decide to leave the check for more than one probe, the logic around deciding if any are healthy or not is probably wrong. It;s currently iterating all probes, setting `allunhealthy` to true if any are healthy, but only using the health of the last probe in the list when setting the value in `gwAddressHealth`

Reply:
Given how we are currently constructing the name if there were more than one I would expect that to be a bug https://github.com/Kuadrant/multicluster-gateway-controller/blob/main/pkg/controllers/dnspolicy/dnspolicy_healthchecks.go#L132